server/cmwell-ws/src/pack/logback.xml

```diff
@@ -1,87 +1,122 @@
+        <!-- this quadruples logging throughput -->
+        <immediateFlush>false</immediateFlush>
     </appender>
 
-    <appender name="x-fix" class="ch.qos.logback.core.FileAppender">
-        <file>${application.home:-.}/logs/x-fix.log</file>
-        <append>true</append>
-        <layout class="ch.qos.logback.classic.PatternLayout">
-            <pattern>%d{yyyy-MM-dd HH:mm:ss}, %p, %c, %t, %L, %C{1}, %M %m%n</pattern>
-        </layout>
+    <appender name="ASYNC_ACCESS_FILE" class="ch.qos.logback.classic.AsyncAppender">
+        <appender-ref ref="ACCESS_FILE" />
+        <maxFlushTime>0</maxFlushTime>
+        <discardingThreshold>0</discardingThreshold>
     </appender>
 
-    <appender name="CW_APP" class="ch.qos.logback.core.rolling.RollingFileAppender">
-        <file>logs/cw-application.log</file>
-        <append>true</append>
-        <layout class="ch.qos.logback.classic.PatternLayout">
-            <pattern>%d{yyyy-MM-dd HH:mm:ss}, %p, %c, %t, %L, %C{1}, %M %m%n</pattern>
-        </layout>
-
-        <rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
-            <fileNamePattern>logs/cw-application.log.%i</fileNamePattern>
-            <maxIndex>21</maxIndex>
+    <appender name="X_FIX" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>${application.home:-.}/logs/x-fix.log</file>
+        <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
+            <!-- daily rollover with compression -->
+            <fileNamePattern>${application.home:-.}/logs/x-fix-log-%d{yyyy-MM-dd, UTC}.%i.gz</fileNamePattern>
+            <!-- keep 1 week worth of history (max 20GB) with each file compressed after 100MB -->
+            <maxFileSize>100MB</maxFileSize>
+            <maxHistory>7</maxHistory>
+            <totalSizeCap>20GB</totalSizeCap>
+            <!-- in case of process terminating too early for rollover - do the rollover during start -->
+            <cleanHistoryOnStart>true</cleanHistoryOnStart>
         </rollingPolicy>
-        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
-            <maxFileSize>10000000</maxFileSize>
-        </triggeringPolicy>
+        <encoder>
+            <pattern>%date{ISO8601} %-44.44([%thread]) %-5level %-36.36logger{36} - %msg%n</pattern>
+        </encoder>
+        <!-- this quadruples logging throughput -->
+        <immediateFlush>false</immediateFlush>
     </appender>
 
-    <logger name="cmwell.xfix" level="INFO" additivity="true">
-        <appender-ref ref="x-fix"/>
-    </logger>
+    <appender name="ASYNC_X_FIX" class="ch.qos.logback.classic.AsyncAppender">
+        <appender-ref ref="X_FIX" />
+        <maxFlushTime>0</maxFlushTime>
+        <discardingThreshold>0</discardingThreshold>
+    </appender>
 
-    <logger name="access" level="INFO" additivity="false">
-        <appender-ref ref="ACCESS_FILE" />
+    <logger name="cmwell.xfix" level="INFO" additivity="false">
```

Comment:
lets document why x-fix log get's special treatment, and is set to minimum INFO logging level